### PR TITLE
8275185: Remove dead code and clean up jvmstat LocalVmManager

### PR DESCRIPTION
**Pull request by iklam**
LocalVmManager and PerfDataFile have APIs that are supposed to look for VMs owned by a specific user. No one uses these APIs, and they don't work anyway.

The current code is very confusing to look at. Since we're likely to change code in this area for further container support, it's better to clean up the code now.

- Remove all APIs that take a user name
- Also removed PerfDataFile.getFile() methods that are unused
- Cleaned up the code that looks up the hsperfdata_xxx files
  - Fix comments to explain what's happening
  - Avoid using Matcher.reset which is not thread-safe
  - Renamed confusing variables such as `userFilter` to make the code more readable
  - LocalVmManager.activeVms() probably doesn't need to be synchronized, but I kept it anyway to avoid unnecessary risks.

Testing with Oracle CI: tiers1-4, plus hs-tier5-rt (which tests containers and have extensive use of the management tools).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8275185](https://bugs.openjdk.java.net/browse/JDK-8275185): Remove dead code and clean up jvmstat LocalVmManager

### Reviewers
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**)
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**)
 * [Kevin Walls](https://openjdk.java.net/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5923/head:pull/5923` \
`$ git checkout pull/5923`

Update a local copy of the PR: \
`$ git checkout pull/5923` \
`$ git pull https://git.openjdk.java.net/jdk pull/5923/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5923`

View PR using the GUI difftool: \
`$ git pr show -t 5923`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5923.diff">https://git.openjdk.java.net/jdk/pull/5923.diff</a>

</details>

### CODE REVIEW

File: src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java [hunk 1]
```diff
@@ -35,142 +35,83 @@
...
-                        for (int j = 0; j < files.length; j++) {
-                            if (files[j].isFile() && files[j].canRead()) {
-                                int vmid = PerfDataFile.getLocalVmId(files[j]);
-                                if (vmid != -1) {
-                                  jvmSet.add(vmid);
-                                }
-                            }
-                        }
-                    }
+            // 1.4.2 and later: Look for the files {tmpdir}/hsperfdata_{any_user_name}/[0-0]+
```
[the remark this answers is not in the document]
Comment (iklam) [reply src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java]: Fixed.

File: src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java [hunk 2]
```diff
@@ -35,142 +35,83 @@
...
+        userDirFileFilter = new FilenameFilter() {
+            public boolean accept(File dir, String name) {
+                return userDirFilePattern.matcher(name).matches();
+            }
+        };
 
-        tmpFileFilter = new FilenameFilter() {
+        // 1.4.1 (or earlier?): the files are stored directly under {tmpdir}/ with
+        // the following pattern.
+        Pattern oldtmpFilePattern = Pattern.compile("^hsperfdata_[0-9]+(_[1-2]+)?$");
```
[the remark this answers is not in the document]
Comment (iklam) [reply src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java]: I restored this line to use PerfDataFile.tmpFileNamePattern, as before my changes. Yes, that's an odd way of naming a file.

**Comment by iklam** [comment]
> Hi Ioi, It looks good to me except the line 105 commented by Chris. I wonder how should this be tested to make sure there are no regressions. Do we really care about pre 1.4.2 formats? If so, what is the way to test it? We have a little lack of knowledge in this area. I guess, we need the performance team to get involved into this review. Thanks, Serguei

I am not sure about 1.4.2 formats. I guess we don't really care, but we should probably remove it in a separate RFE. I am trying to keep this PR from having any functional changes, and will just remove dead code.

For testing, I am running tiers 1-4. Will that be enough?

I'll ping the performance team.

Thanks
- Ioi

**Comment by iklam** [comment]
> Nice cleanup - although was it not better for PerfDataFile.java to "own" the definitions of what a perfdata filename looks like? That might be what Chris was hinting at. There isn't really that _much_ flipping between two files. 8-)

Hi Kevin, thanks for the review. I've moved the filename patterns back to PerfDataFile.java as you suggested.

**Review by plummercj — APPROVED** [review]

**Review by cl4es — APPROVED** [review]
Supporting 1.4.1 hsperfdata seems pointless at this point, so I support a removal of that code. It also seems reasonable to do as a follow-up, as @iklam suggests, rather than shifting the scope this PR.

**Comment by kevinjwalls** [comment]
> ...moved the filename patterns back to PerfDataFile.java...

Thanks, looks good!

**Review by kevinjwalls — APPROVED** [review]

**Comment by iklam** [comment]
Thanks @plummercj  @kevinjwalls @cl4es for the review. I filed https://bugs.openjdk.java.net/browse/JDK-8276687 to remove the JDK 1.4.1 support.
Passed Oracle CI tiers1-4
/integrate

**Comment by sspitsyn** [comment]
Hi Ioi,
Sorry for being late here.
Just to complete this...
Thank you for your answers! I'm okay with them.

**Comment by iklam** [comment]
> Hi Ioi,
> Sorry for being late here.
> Just to complete this...
> Thank you for your answers! I'm okay with them.

Thanks Serguei!

**Pull request closed without merging**